stl/inc/source_location

```diff
@@ -0,0 +1,64 @@
+// source_location standard header (core)
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _SOURCE_LOCATION_
+#define _SOURCE_LOCATION_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+#if !_HAS_CXX20
+#pragma message("The contents of <source_location> are available only with C++20 or later.")
+#else // ^^^ !_HAS_CXX20 / _HAS_CXX20 vvv
+
+#include <cstdint>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+struct source_location {
+    static constexpr source_location current(uint_least32_t _InLine = __builtin_LINE(),
+        uint_least32_t _InColumn = __builtin_COLUMN(), const char* _InFile = __builtin_FILE(),
+        const char* _InFunction = __builtin_FUNCTION()) noexcept {
+        source_location _Result;
+        _Result._Line     = _InLine;
+        _Result._Column   = _InColumn;
+        _Result._File     = _InFile;
+        _Result._Function = _InFunction;
+        return _Result;

```

Comment:
@StephanTLavavej [INDENTATION STYLE RANT] Theres some ugly indentation here. I will suggest a lecture by Kavlin Henney: ITT 2016 - Kevlin Henney - Seven Ineffective Coding Habits of Many Programmers https://www.youtube.com/watch?t=1263&v=ZsHMHukIlJY [/INDENTATION STYLE RANT] 

The below formatting, I believe I get by using clang-format parameters (https://clang.llvm.org/docs/ClangFormatStyleOptions.html):
_AllowAllConstructorInitializersOnNextLine: false
AllowAllParametersOfDeclarationOnNextLine: false
BinPackArguments: false
BinPackParameters: false
AlignConsecutiveAssignments: AcrossComments
AlignConsecutiveDeclarations: AcrossComments_

```
static constexpr source_location current(uint_least32_t _InLine     = __builtin_LINE(),
                                         uint_least32_t _InColumn   = __builtin_COLUMN(),
                                         const char*    _InFile     = __builtin_FILE(),
                                         const char*    _InFunction = __builtin_FUNCTION()) noexcept
{
    source_location _Result;
    _Result._Line     = _InLine;
    _Result._Column   = _InColumn;
    _Result._File     = _InFile;
    _Result._Function = _InFunction;
    return _Result;
}
```

As style is a contentious issue, please feel free to ignore this rant.